components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -701,7 +701,7 @@ private void generateJWTAndAddToTransportHeaders(MessageContext messageContext,
         String webAppContext = (String) messageContext.getProperty(RESTConstants.REST_API_CONTEXT);
         String webAppVersion = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API_VERSION);
 
-        String jwtToken = AppMDAO.jwtGenerator.generateToken(samlAttributes, webAppContext, webAppVersion);
+        String jwtToken = AppMDAO.getTokenGenerator().generateToken(samlAttributes, webAppContext, webAppVersion);
```

Comment:
Better use AppManagerConfiguration to get the configs and load the reference to TokenGenerate in the handler init method.